bindings/nodejs/src/message_handler.rs

```diff
@@ -166,6 +166,32 @@ pub fn listen(mut cx: FunctionContext) -> JsResult<JsUndefined> {
     Ok(cx.undefined())
 }
 
+pub fn clear_listeners(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+    let js_arr_handle: Handle<JsArray> = cx.argument(0)?;
+    let vec: Vec<Handle<JsValue>> = js_arr_handle.to_vec(&mut cx)?;
+    let mut event_types = vec![];
+    for event_string in vec {
+        let event_type = event_string.downcast::<JsString, FunctionContext>(&mut cx).unwrap();
+        let wallet_event_type = WalletEventType::try_from(event_type.value(&mut cx).as_str()).unwrap();
```

Comment:
Are these 2 unwraps safe ? If yes, a little comment on why you are 100% sure they will never panic would be greatly appreciated

Reply:
They can panic, but neon will convert it to throws or whatever it's called on the nodejs side
```JS
        try {
            manager.listen(['invalid event'], callback2);
        } catch (e) {
            console.log(e)
        }

        const address = await account.generateAddresses(1);
        console.log('New address:', address);
```
Would then return 
```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: "invalid event type invalid event"', src/message_handler.rs:145:95
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: internal error in Neon module: called `Result::unwrap()` on an `Err` value: "invalid event type invalid event"
    at MessageHandler.listen (/home/t/Desktop/wallet.rs/bindings/nodejs/out/lib/MessageHandler.js:30:38)
    at AccountManager.listen (/home/t/Desktop/wallet.rs/bindings/nodejs/out/lib/AccountManager.js:120:36)
    at run (/home/t/Desktop/wallet.rs/bindings/nodejs/examples/7-events.js:34:21)
New address: [
  {
    address: 'rms1qq3dggwaaegezpelekdxp6jumzkzusyyuv76lefk2m25pmvr02clqj0ne07',
    keyIndex: 2,
    internal: false,
    used: false
  }
]
```
which is not a nice error message, but would still work

Reply:
So I played a bit around and
```
        let event_type = event_string.downcast_or_throw::<JsString, FunctionContext>(&mut cx)?;
        let wallet_event_type =
            WalletEventType::try_from(event_type.value(&mut cx).as_str()).or_else(|e| cx.throw_error(e))?;
```
works fine, @oddgrd can you update in your code and at the other place?

Reply:
I will update them, thanks for investigating Thoralf!